wallet-web/components/bond/BondNodeForm.tsx

```diff
@@ -1,380 +1,480 @@
-                sphinx_key: event.target.sphinxkey.value,
-                identity_key: event.target.identity.value,
-                version: event.target.version.value,
-                location: event.target.location.value
-            }
-        }
+  }
+
+  const constructGatewayBondingInfo = (
+    data: TFormInput
+  ): BondingInformation => {
+    return {
+      amount: data.amount,
+      nodeDetails: {
+        host: data.host,
+        mix_port: advancedShown ? parseInt(data.mixPort) : DEFAULT_MIX_PORT,
+        clients_port: advancedShown
+          ? parseInt(data.clientsPort)
+          : DEFAULT_CLIENTS_PORT,
+        sphinx_key: data.sphinxKey,
+        identity_key: data.identityKey,
+        version: data.version,
+        location: data.location,
+      },
     }
-
-    const submitForm = (event: any) => {
-        event.preventDefault()
-
-        if (validateForm(event)) {
-            if (props.type == NodeType.Mixnode) {
-                return props.onSubmit(constructMixnodeBondingInfo(event))
-            } else {
-                return props.onSubmit(constructGatewayBondingInfo(event))
-            }
-        }
+  }
+
+  const submitForm = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    const target = event.target as unknown as TFormInput
+
+    if (validateForm(target)) {
+      if (props.type == NodeType.Mixnode) {
+        return props.onSubmit(constructMixnodeBondingInfo(target))
+      } else {
+        return props.onSubmit(constructGatewayBondingInfo(target))
+      }
     }
+  }
+
+  let minimumBond = props.minimumMixnodeBond
+  if (props.type == NodeType.Gateway) {
+    minimumBond = props.minimumGatewayBond
+  }
+
+  // if this whole interface wasn't to be completely redone in a month time, I would have definitely redone the form
+  // but I guess it's fine for time being
+  return (
+    <form onSubmit={submitForm}>
+      <Grid container spacing={3}>
+        <Grid item xs={12} sm={8}>
+          <TextField
+            required
+            id='amount'
+            name='amount'
+            label={`Amount to bond (minimum ${nativeToPrintable(
+              minimumBond.amount
+            )} ${minimumBond.denom})`}
+            error={!validity.validAmount}
+            {...(!validity.validAmount
+              ? {
+                  helperText: `Enter a valid bond amount (minimum ${nativeToPrintable(
+                    minimumBond.amount
+                  )})`,
+                }
+              : {})}
+            fullWidth
+            InputProps={{
+              endAdornment: (
+                <InputAdornment position='end'>{DENOM}</InputAdornment>
+              ),
+            }}
+            onChange={(e: ChangeEvent<HTMLInputElement>) => {
+              if (
+                typeof parseInt(e.target.value) === 'number' &&
+                parseInt(accountBalance.amount) - parseInt(e.target.value) < 1
```

Comment:
Could we change that to instead show the warning if after bonding user is going to have less than 2 times the gas fee? (two because 1 is required for bonding and the other one for potential unbonding)
Also, maybe this check should be moved to another function? It's getting slightly busy in this element

Reply:
I meant component*

Reply:
I moved some of the logic into a helper function. You're right, it makes sense to do it that way. It also means the function can be reused for the delegation form.